cloudformation/inUse.yaml

```diff
@@ -0,0 +1,195 @@
+              - arn:aws:s3:::composer-dist/*
+      Roles:
+        - !Ref 'ServerRole'
+  SSHSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Allow SSH access from the office
+      VpcId: !Ref 'VpcId'
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: '22'
+          ToPort: '22'
+          CidrIp: 10.0.0.0/8
+  AppServerSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: application servers
+      VpcId: !Ref 'VpcId'
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 9000
+          ToPort: 9000
+          SourceSecurityGroupId: !Ref 'LoadBalancerSecurityGroup'
+  LoadBalancerSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: load balancer
+      VpcId: !Ref 'VpcId'
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 443
+          ToPort: 443
+          CidrIp: 77.91.248.0/21
+  LoadBalancer:
+    Type: AWS::ElasticLoadBalancing::LoadBalancer
+    Properties:
+      SecurityGroups:
+        - !GetAtt 'LoadBalancerSecurityGroup.GroupId'
+      CrossZone: true
+      Subnets: !Ref 'PublicVpcSubnets'
+      Listeners:
+        - LoadBalancerPort: '443'
+          InstancePort: '9000'
+          Protocol: HTTPS
+          SSLCertificateId: !Ref 'CertificateArn'
+      HealthCheck:
+        Target: HTTP:9000/management/healthcheck
+        HealthyThreshold: '2'
+        UnhealthyThreshold: '2'
+        Interval: '10'
+        Timeout: '5'
+      Tags:
+        - Key: Stage
+          Value: !Ref 'Stage'
+        - Key: Stack
+          Value: !Ref 'Stack'
+        - Key: App
+          Value: 'in-use'
+  AutoscalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      VPCZoneIdentifier: !Ref 'PrivateVpcSubnets'
+      LaunchConfigurationName: !Ref 'LaunchConfig'
+      LoadBalancerNames:
+        - !Ref 'LoadBalancer'
+      MinSize: 1
+      MaxSize: 2
+      Cooldown: '180'
+      HealthCheckType: ELB
+      HealthCheckGracePeriod: 300
+      Tags:
+        - Key: Name
+          Value: !Join ['', [!Ref 'Stage', ':in-use']]
+          PropagateAtLaunch: 'true'
+        - Key: Stage
+          Value: !Ref 'Stage'
+          PropagateAtLaunch: 'true'
+        - Key: Stack
+          Value: !Ref 'Stack'
+          PropagateAtLaunch: 'true'
+        - Key: App
+          Value: in-use
+          PropagateAtLaunch: 'true'
+  LaunchConfig:
+    Type: AWS::AutoScaling::LaunchConfiguration
+    Properties:
+      KeyName: !Ref 'KeyName'
+      ImageId: !Ref 'AMI'
+      SecurityGroups:
+        - !Ref 'SSHSecurityGroup'
+        - !Ref 'AppServerSecurityGroup'
+      InstanceType: !Ref 'InstanceType'
+      IamInstanceProfile: !Ref 'ServerInstanceProfile'
+      UserData:
+        'Fn::Base64': !Sub |
+          #!/bin/bash -ev
+          aws s3 cp s3://composer-dist/flexible/${Stage}/in-use/in-use.deb /tmp
+          sleep 30
```

Comment:
I know having a sleep here is really strange, but we get a race condition with dpkg still being in use when the instance first spins up. Seems no other apps hit this problem, because most of our .debs are double the size of inUse